java/bulk-export/bulk-export-lambda/src/main/java/sleeper/bulkexport/lambda/SqsBulkExportProcessorLambda.java

```
@@ -0,0 +1,146 @@
+     */
+    public SqsBulkExportProcessorLambda(AmazonSQS sqsClient, AmazonS3 s3Client, AmazonDynamoDB dynamoClient)
+            throws ObjectFactoryException {
+        this.sqsClient = sqsClient;
+        this.s3Client = s3Client;
+        this.dynamoClient = dynamoClient;
+
+        String bucket = System.getenv(CONFIG_BUCKET.toEnvironmentVariable());
+        instanceProperties = S3InstanceProperties.loadFromBucket(s3Client, bucket);
+        TablePropertiesProvider tablePropertiesProvider = S3TableProperties.createProvider(instanceProperties, s3Client,
+                dynamoClient);
+        bulkExportQuerySerDe = new BulkExportQuerySerDe();
+        processor = SqsBulkExportProcessor.builder()
+                .sqsClient(sqsClient)
+                .s3Client(s3Client)
+                .dynamoClient(dynamoClient)
+                .instanceProperties(instanceProperties)
+                .tablePropertiesProvider(tablePropertiesProvider)
+                .build();
+    }
+
+    /**
+     * Handles the incoming SQS event, processes each message, and performs the bulk
+     * export operation.
+     *
+     * @param event   The SQS event containing the messages to be processed.
+     * @param context The AWS Lambda context object providing runtime information.
+     * @return Always returns null.
+     */
+    @Override
+    public Void handleRequest(SQSEvent event, Context context) {
+        event.getRecords().stream()
+                .map(SQSEvent.SQSMessage::getBody)
+                .peek(body -> LOGGER.debug("Received message with body {}", body))
+                .flatMap(body -> deserialiseAndValidate(body).stream())
+                .peek(json -> LOGGER.debug(json.toString()))
+                .forEach(query -> {
+                    try {
+                        processor.processExport(query);
+                    } catch (ObjectFactoryException e) {
+                        LOGGER.error("Failed to process export query", e);
```

I'm not sure we want to discard this failure? We could throw the exception all the way out, or report just the ones that failed with SQSBatchResponse. Either way would retry the messages or send them to the dead letter queue, whereas as it is these failures are just ignored.

---

I can also see sending this directly to the dead letter queue, though there are some cases where this exception is retryable. It might be best to assume we need to retry this, and maybe consider splitting up the exception.